integration/dbfs/test_integration.py

```diff
@@ -0,0 +1,124 @@
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint:disable=redefined-outer-name
+import os
+
+import pytest
+
+from databricks_cli.dbfs import cli
+from databricks_cli.dbfs.dbfs_path import DbfsPath
+from tests.utils import invoke_cli_runner
+
+DBFS_TEST_PATH = 'dbfs:/databricks-cli-test'
+LOCAL_TEMP_FILE = 'temp-file.txt'
+LOCAL_TEMP_DIR = 'temp-dir'
+LOCAL_TEST_FILE = 'test-file.txt'
+LOCAL_TEST_DIR = 'test-dir'
+LOCAL_TEST_FILE_IN_DIR = os.path.join(LOCAL_TEST_DIR, LOCAL_TEST_FILE)
+TEST_FILE_CONTENTS = 'Hi I am a test file.\n'
+
+
+@pytest.fixture()
+def dbfs_dir():
+    invoke_cli_runner(cli.mkdirs_cli, [DBFS_TEST_PATH])
+    yield
+    invoke_cli_runner(cli.rm_cli, ['-r', DBFS_TEST_PATH])
+
+
+@pytest.fixture()
+def local_dir(tmpdir):
+    """
+    Creates a tmpdir with this structure.
+    - tmpdir
+      - test-file.txt
+      - test-dir
+        - test-file-in-dir.txt
+    """
+    path = tmpdir.strpath
+    with open(os.path.join(path, LOCAL_TEST_FILE), 'wt') as f:
+        f.write(TEST_FILE_CONTENTS)
+    os.mkdir(os.path.join(path, LOCAL_TEST_DIR))
+    with open(os.path.join(path, LOCAL_TEST_FILE_IN_DIR), 'wt') as f:
+        f.write(TEST_FILE_CONTENTS)
+    yield tmpdir
+
+
+def assert_local_file_exists(local_path, expected_contents):
+    assert os.path.exists(local_path)
+    with open(local_path, 'rt') as f:
+        assert f.read() == expected_contents
+
+
+def assert_dbfs_file_exists(dbfs_path):
+    basename = dbfs_path.basename
+    dirname = dbfs_path.absolute_path[:-len(basename)]
+    res = invoke_cli_runner(cli.ls_cli, [dirname])
```

Comment:
`dbfs ls` the absolute path should work and return the basename.

Reply:
The current way it's implemented works for both asserting files/directories. `dbfs ls` on the absolute path would only work on asserting files.